src/lib/util/cursor.c

```diff
@@ -402,6 +402,56 @@ void fr_cursor_merge(fr_cursor_t *cursor, fr_cursor_t *to_append)
 	}
 }
 
+/** Return the first item that satisfies an evaluation function.
+ *
+ * @param[in] cursor	to operate on
+ * @param[in] eval	evaluation function
+ * @param[in] uctx	context for the evaluation function
+ * @return the first item satisfying eval, or NULL if no such item exists
+ */
+void *fr_cursor_filter_head(fr_cursor_t *cursor, fr_cursor_eval_t eval, void *uctx)
+{
+	void *item;
+
+	item = fr_cursor_head(cursor);
+	if (eval(item, uctx)) return item;
+
+	return fr_cursor_filter_next(cursor, eval, uctx);
+}
+
+/** Return the next item, skipping the current item, that satisfies an evaluation function.
+ *
+ * @param[in] cursor	to operate on
+ * @param[in] eval	evaluation function
+ * @param[in] uctx	context for the evaluation function
+ * @return the next item satisfying eval, or NULL if no such item exists
+ */
+void *fr_cursor_filter_next(fr_cursor_t *cursor, fr_cursor_eval_t eval, void *uctx)
+{
+	void *item;
+
+	do { item = fr_cursor_next(cursor); } while (item && !eval(item, uctx));
```

Comment:
Can you put this on multiple lines?  It's clearer that way

Reply:
Will do.